packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -1147,7 +1096,7 @@ class SlideshowView extends GalleryComponent {
         style={galleryStyle}
       >
         {this.createDebugMsg()}
-        {this.createNavArrows()}
+        {shouldRenderNavArrows(getShouldRenderArrowsArgs(this.props)) && this.createNavArrows()}
```

Comment:
`getShouldRenderArrowsArgs` can be turned to a private method of the `navigationArrowUtils` you created. When a module (or even file)  such as `navigationArrowUtils` contains fewer exports, it makes it easier to understand the dependencies of the module. Private methods are awesome (private === not exported).
So, here's what you can do: 

1. Pass `container` and `options` to `shouldRenderNavArrows`
2. Pass `options` to `getShouldRenderArrowsArgs`
3. Remove the `export` of `getShouldRenderArrowsArgs` - since it's not used elsewhere
4. Celebrate

Reply:
Great explanation, thanks!
I've turned getShouldRenderArrowsArgs to private method, mind taking a look?
Please note that shouldRenderNavArrows must take a few more arguments (other than options and container), therefore some of the deconstruction happens now in slideshowView. I think it's a tradeoff (turning getShouldRenderArrowsArgs private <-> amount of deconstructing in slideshowView) I can't avoid, do you have an idea on how to avoid it? does the current change good enough?
thanks again for the explanation!
@nirnaor

Reply:
As a rule of thumb I try to pass the minimum amount of dependencies (arguments in your case) from the top level to the bottom (from `sliedeshowView` to `navigationArrowUtils` in your case). Therefor, destructing at the higher level (`slideshowview`) **is better** (like you did). So I think you are good on that front.